python_modules/libraries/dagster-airlift/dagster_airlift/core/components/airflow_instance/component.py

```diff
@@ -106,22 +114,51 @@ def scaffold(self, request: ScaffoldRequest[AirflowInstanceScaffolderParams]) ->
         scaffold_component(request, full_params)
 
 
-def resolve_auth(context: ResolutionContext, model) -> AirflowAuthBackend:
-    if model.auth.type == "basic_auth":
+def resolve_auth(context: ResolutionContext, auth: BaseModel) -> AirflowAuthBackend:
+    if auth.type == "basic_auth":
+        resolved_model = resolve_fields(auth, ResolvedAirflowBasicAuthBackend, context)
         return AirflowBasicAuthBackend(
-            webserver_url=model.auth.webserver_url,
-            username=model.auth.username,
-            password=model.auth.password,
+            webserver_url=resolved_model["webserver_url"],
+            username=resolved_model["username"],
+            password=resolved_model["password"],
+        )
+    elif auth.type == "mwaa":
+        resolved_model = resolve_fields(auth, ResolvedAirflowMwaaAuthBackend, context)
```

Comment:
nit: maybe call this "resolved_model_fields" to make it more obvious that this is a dictionary, not a model